src/main/java/org/ngafid/routes/GetUserEmailPreferences.java

```diff
@@ -45,38 +56,27 @@ public Object handle(Request request, Response response) {
 
         LOG.info("handling " + this.getClass().getName() + " route");
 
+        //Unpack Fetching Data
+        String handleFetchType = request.queryParams("handleFetchType");
+
         final Session session = request.session();
         User sessionUser = session.attribute("user");
 
-        String userIDParam = request.queryParams("userID");
-        int userID;
-
-        //No userID parameter specified, try using the session user
-        if (userIDParam == null) {
 
-            if (sessionUser == null) {
-                response.status(400);
-                return gson.toJson(new ErrorResponse("[GET Email Preference Error]", "Missing userID parameter + no session user available"));
-            }
-
-            userID = sessionUser.getId();
-
-        }
+        int fleetUserID = -1;
         
-        //userID was specified, try to parse it
-        else {
-
-            try {
-                userID = Integer.parseInt(userIDParam);
-            } catch (NumberFormatException e) {
-                response.status(400);
-                return gson.toJson(new ErrorResponse("[GET Email Preference Error]", "Specified an invalid userID parameter"));
-            }
+        //Fetching Session User...
+        if (handleFetchType.equals("HANDLE_FETCH_USER")) {
+            fleetUserID = sessionUser.getId();
+        }
 
+        //Fetching a Manager's Fleet User...
+        else if (handleFetchType.equals("HANDLE_FETCH_MANAGER")) {
```

Comment:
Validate that the current use has these permissions / is a manager

Reply:
I don't think this was addressed -- you will have to take the user id of the current user and validate that they are in fact a manager.